orchagent/monitorTXorch.cpp

```diff
@@ -0,0 +1,300 @@
+                m_poolingPeriod = stoi(value);
+                auto interv = timespec { .tv_sec = m_poolingPeriod, .tv_nsec = 0 };
+                m_timer->setInterval(interv);
+                m_timer->reset();
+            }
+        }
+
+    // change threshold val
+    } else if (key == THRESHOLD_KEY) {
+
+        for (auto element : data) {
+            const auto &field = fvField(element);
+            const auto &value = fvValue(element);
+
+            if (field == "value") {
+                SWSS_LOG_NOTICE("changing threshold to %s", value.c_str());
+                m_threshold = stoi(value);
+            }
+
+        }
+    
+    } else {
+        SWSS_LOG_ERROR("unknown command");
+        return false;
+    }
+    return true;
+
+}
+
+void MonitorTXOrch::doTask(Consumer &consumer) {
+    SWSS_LOG_ENTER();
+    SWSS_LOG_NOTICE("inside dotask - consumer");
+
+    /* check if needed */
+    string table_name = consumer.getTableName();
+    if (table_name != CFG_TX_ERROR_TABLE_NAME) {
+        SWSS_LOG_ERROR("Invalid table name - %s", table_name.c_str());
+    }
+
+    auto it = consumer.m_toSync.begin();
+    while(it != consumer.m_toSync.end()) {
+        KeyOpFieldsValuesTuple t = it->second;
+
+        string key = kfvKey(t);
+        string op = kfvOp(t);
+        vector<FieldValueTuple> fieldValues = kfvFieldsValues(t);
+        
+        if (op == SET_COMMAND) 
+        {
+            bool isSet = handleSetCommand(key, fieldValues);
+            if (!isSet) {
+                SWSS_LOG_ERROR("set has not succeed");
+            }
+        } 
+        else 
+        {
+            SWSS_LOG_ERROR("Unknown operation type %s", op.c_str());
+        }
+
+        consumer.m_toSync.erase(it++);
+
+    }
+}
+
+// void MonitorTXOrch::doTask(Consumer &consumer) {
+//     SWSS_LOG_ENTER();
+//     SWSS_LOG_NOTICE("implementation for now ....");
+// }
+
+
+
+bool MonitorTXOrch::getTXStatistics() {
+    SWSS_LOG_ENTER();
+    SWSS_LOG_NOTICE("inside get statistics");
+
+    for (auto &entry : m_portsStringsMap) {
+        SWSS_LOG_NOTICE("inside for loop - port  ... ");
+
+        sai_object_id_t currPortID = entry.first;
+        if (!getTXCountersById(currPortID)){
+            return false;
+        }
+    }
+    return true;
+
+}
+
+/*
+ * get tx counters from db by port id 
+ * 
+ */
+bool MonitorTXOrch::getTXCountersById(sai_object_id_t portID) {
+    SWSS_LOG_ENTER();
+
+    string strValue;
+    string oidStr = m_portsStringsMap.find(portID)->second;
+
+    SWSS_LOG_NOTICE("getting tx-err counters to specific port %s", oidStr.c_str());
+
+    if (!m_countersTable->hget(oidStr, counterName, strValue)) {
+        SWSS_LOG_NOTICE("Error reading counters table");
+        // insert unknown state to DB
+
+        vector<FieldValueTuple> fieldValuesVector;
+        fieldValuesVector.emplace_back(STATE_DB_TX_STATE, "UNKNOWN");
+        m_stateTxErrTable.set(oidStr, fieldValuesVector);
```

Comment:
there are two places where you set the state, can you have a small routine that sets the state so this part of code will be reused.